languages/wikipron/src/whitelist.py

```diff
@@ -1,45 +1,43 @@
 #!/usr/bin/env python
+"""Filters TSVs through a whitelist file.
 
-import sys
+This script takes lines from a source TSV and writes only lines with
+pronunciations containing characters found in the whitelist file to a new TSV."""
+
+import argparse
 import re
+import sys
 
+from typing import FrozenSet, Iterator
 
-def whitelist_reader(path):
-    """Reads in whitelist file"""
+
+def whitelist_reader(path: str) -> Iterator[str]:
+    """Reads in whitelist file."""
     with open(path, "r") as source:
         for line in source:
-            yield line.rstrip().split()[0]
-
-
-def filter(tsv_path, phonemes, output_path):
-    """Creates tsv filtered by whitelist"""
-    with open(tsv_path, "r") as source:
-        with open(output_path, "w") as output:
-            for line in source:
-                pron = set(line.split("\t")[1].split())
-                if phonemes.issuperset(pron):
-                    print(line.rstrip(), file=output)
-                else:
-                    pass
-
-
-def main() -> None:
-    tsv_path = sys.argv[1]
-    iso639_code = tsv_path[tsv_path.rindex("/") + 1:tsv_path.index("_")]
-    dialect = re.search("_(.*)_", tsv_path)
-    if dialect:
-        whitelist_path = f"../whitelist/{iso639_code}_{dialect[1]}_phonemic.whitelist"
-        output_path = f"../tsv/{iso639_code}_{dialect[1]}_filtered_phonemic.tsv"
-    else:
-        whitelist_path = f"../whitelist/{iso639_code}_phonemic.whitelist"
-        output_path = f"../tsv/{iso639_code}_filtered_phonemic.tsv"
-    try:
-        whitelist_phonemes = set(whitelist_reader(whitelist_path))
-        whitelist_phonemes.add(" ")
-        filter(tsv_path, whitelist_phonemes, output_path)
-    except FileNotFoundError:
-        pass
+            # Removes comments from line.
+            line = re.sub("\s.*", "", line)
```

Comment:
This will just delete everything after space. But that's not exactly how a comment works. A comment is defined as optional space `\s*` mandatorily followed by `#` then the rest of the line `.*$`; I think that would be a bit more constrained.

Reply:
Okay, that makes sense, should I push this change now?

Reply:
Please make that change, test, and then add, commit, and push the change. Then you can press the big green "Squash and merge" button" (I think you should have the privileges to do that), or you can ping me and I'll do it.